pandas/core/frame.py

```diff
@@ -4870,14 +4870,24 @@ def combine(self, other, func, fill_value=None, overwrite=True):
                 series[this_mask] = fill_value
                 otherSeries[other_mask] = fill_value
 
-            # if we have different dtypes, possibly promote
-            new_dtype = this_dtype
-            if not is_dtype_equal(this_dtype, other_dtype):
```

Comment:
any way to simplify this, adding this condition is a bit odd; if these are not the same dtype then we will astype anyhow, why does adding this extra check help here?

Reply:
Ah! This line is none-sense because `find_common_type` exists in the next line!